src/Xamarin.Android.Build.Tasks/Resources/MultiDexLoader.Debug.java

```diff
@@ -0,0 +1,512 @@
+	
+	@Override
+	public boolean onCreate ()
+	{
+		return true;
+	}
+
+	@Override
+	public void attachInfo (android.content.Context context, android.content.pm.ProviderInfo info) {
+		mIncrementalDeploymentDir = getIncrementalDeploymentDir (context);
+		String externalResourceFile = getExternalResourceFile ();
+
+		File codeCacheDir = context.getCacheDir ();
+		String nativeLibDir = context.getApplicationInfo ().nativeLibraryDir;
+		String dataDir = context.getApplicationInfo ().dataDir;
+		String packageName = context.getPackageName ();
+
+		List<String> dexes = getDexList (packageName);
+		if (dexes != null && dexes.size () > 0) {
+			IncrementalClassLoader.inject (
+				MultiDexLoader.class.getClassLoader (),
+				packageName,
+				codeCacheDir,
+				nativeLibDir,
+				dexes);
+		}
+		monkeyPatchExistingResources (context, externalResourceFile, getActivities (context, false));
+		super.attachInfo (context, info);
+	}
+
+
+	private String mIncrementalDeploymentDir;
+
+	
+	private static String getIncrementalDeploymentDir (Context context)
+	{
+		// For initial setup by Seppuku, it needs to create the dex deployment directory at app bootstrap.
+		// dex is special, important for mono runtime bootstrap.
+		String dir = new File (
+			android.os.Environment.getExternalStorageDirectory (),
+			"Android/data/" + context.getPackageName ()).getAbsolutePath ();
+		dir = new File (dir).exists () ?
+			dir + "/files" :
+			"/data/data/" + context.getPackageName () + "/files";
+		String dexDir = dir + "/.__override__/dexes";
+		if (!new File (dexDir).exists ())
+			new File (dexDir).mkdirs ();
+		return dir + "/";
+	}
+
+	private String getExternalResourceFile () {
+		String base = mIncrementalDeploymentDir;
+		String resourceFile = base + ".__override__/packaged_resources";
+		if (!(new File (resourceFile).isFile ())) {
+			resourceFile = base + ".__override__/resources";
+			if (!(new File (resourceFile).isDirectory ())) {
+				Log.v ("MultiDexLoader", "Cannot find external resources, not patching them in");
+				return null;
+			}
+		}
+
+		Log.v ("MultiDexLoader", "Found external resources at " + resourceFile);
+		return resourceFile;
+	}
+
+	private List<String> getDexList (String packageName) {
+		List<String> result = new ArrayList<String> ();
+		String dexDirectory = mIncrementalDeploymentDir + ".__override__/dexes";
+		File[] dexes = new File (dexDirectory).listFiles ();
+		// It is not illegal state when it was launched to start Seppuku
+		if (dexes == null) {
+			return null;
+		} else {
+			for (File dex : dexes) {
+				if (dex.getName ().endsWith (".dex")) {
+					result.add (dex.getPath ());
+				}
+			}
+		}
+
+		return result;
+	}
+
+	private void monkeyPatchExistingResources(Context context,
+                                                    String externalResourceFile,
+                                                    Collection<Activity> activities) {
+        if (externalResourceFile == null) {
+            return;
```

Comment:
This file is an "interesting" mix of tabs and spaces: This method is using spaces, while e.g. `getDexList()` is using tabs.

This file should be normalized to have consistent indenting.